File: src/frontend/src/types/prediction.ts
export type RiskCategory =
  | "LOW"
  | "MEDIUM"
  | "HIGH";

export type AnomalyStatus =
  | "NORMAL"
  | "HIGH";

export type AnomalySeverity =
  | "LOW"
  | "MEDIUM"
  | "HIGH";

export interface FailurePrediction {
  prediction_id: string;
  asset_id: string;
  component_id: string;
  failure_probability: number;
  risk_category: RiskCategory;
  timestamp: string;
}

export interface AnomalyResult {
  asset_id: string;
  component_id: string;
  anomaly_score: number;
  anomaly_status: AnomalyStatus;
  anomaly_severity: AnomalySeverity;
  sensor: string;
  timestamp: string;
}

File: src/backend/member1_backend/app/schemas/prediction.py
from pydantic import BaseModel


class PredictionBase(BaseModel):
    prediction_id: str
    asset_id: str
    component_id: str
    failure_probability: float
    risk_category: str              # LOW | MEDIUM | HIGH
    timestamp: str                  # ISO datetime string


class PredictionCreate(PredictionBase):
    pass


class PredictionResponse(PredictionBase):
    model_config = {"from_attributes": True}
